addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -235,14 +235,15 @@ def _add_invoice_payment_means_nodes(self, document_node, vals):
 
     def _get_address_node(self, vals):
         partner = vals['partner']
-        model = vals.get('model', 'res.partner')
-        country = partner['country' if model == 'res.bank' else 'country_id']
-        state = partner['state' if model == 'res.bank' else 'state_id']
+        country = partner['country' if partner._name == 'res.bank' else 'country_id']
```

Comment:
I must say this does not make any sense. I know it's not your fault, but a partner that is in fact a res.bank will be unreadable by everyone that comes by.